### PR TITLE
Validate that a plan has at most one OutputNode

### PR DESCRIPTION
**Pull request by AlekseiS**

### CODE REVIEW

**Comment by AlekseiS** [comment]
Tests related to `EXPLAIN ANALYZE` fail after the change which is correct, because it currently inserts 2 output nodes.
PR fixing it needs to be merged first: https://github.com/prestodb/presto/pull/6632

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/VerifyOnlyOneOutputNode.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    private static class Visitor
+            extends SimplePlanVisitor<Boolean>
+    {
+        @Override
+        public Void visitOutput(OutputNode node, Boolean context)
+        {
+            if (context) {
+                throw new IllegalStateException("Generated plan has multiple instances of OutputNode");
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/VerifyOnlyOneOutputNode.java]: This should be `IllegalArgumentException` or `VerifyException`. `IllegalStateException` means the method has been "invoked at an illegal or inappropriate time".

Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/VerifyOnlyOneOutputNode.java]:
```java
checkArgument(!context, "Plan has multiple instances of OutputNode");

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/VerifyOnlyOneOutputNode.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    public void validate(PlanNode plan, Session session, Metadata metadata, SqlParser sqlParser, Map<Symbol, Type> types)
+    {
+        plan.accept(new Visitor(), false);
+    }
+
+    private static class Visitor
+            extends SimplePlanVisitor<Boolean>
+    {
+        @Override
+        public Void visitOutput(OutputNode node, Boolean context)
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/VerifyOnlyOneOutputNode.java]: Name this `hasOutputNode`

File: presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/VerifyOnlyOneOutputNode.java (hunk 1 above)
Comment (AlekseiS) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/sanity/VerifyOnlyOneOutputNode.java]: `IllegalStateException` is what other sanity checkers are throwing in this situation, so I used it as well. See `VerifyNoFilteredAggregations` for example.

**Comment by electrum** [comment]
Nit: commit message should be "at most one" since the number is small

**Comment by AlekseiS** [comment]
Addressed the comments and rebased on merged PR for fixing explain analyze.

**Comment by AlekseiS** [comment]
All tests pass now as expected.

**Comment by AlekseiS** [comment]
Rebased.

**Pull request merged**